tests/e2e/e2e.go

```diff
@@ -179,161 +121,151 @@ func Setup(t *testing.T) *TestCtx {
 		t.Logf("Using docker client endpoint: %+v", testCtx.dockerClient.Endpoint())
 	}
 
-	// make sure there are no microservices left from the previous run
-	resetMicroservices(t, testCtx.dockerClient)
-
-	// start the agent
-	assertProcessNotRunning(t, "vpp_agent")
-
-	var agentArgs []string
-
-	if *covPath != "" {
-		e2eCovPath := fmt.Sprintf("%s/%d.out", *covPath, time.Now().Unix())
-		agentArgs = []string{"-test.coverprofile", e2eCovPath}
+	agentImg := vppAgentDefaultImg
+	if img := os.Getenv("VPP_AGENT"); img != "" {
+		agentImg = img
 	}
-	testCtx.agent = startProcess(t, "VPP-Agent", nil, testCtx.outputBuf, testCtx.outputBuf, "/vpp-agent", agentArgs...)
-
-	// prepare HTTP client for access to REST API of the agent
-	httpAddr := fmt.Sprintf(":%d", *agentHTTPPort)
-	testCtx.httpClient = utils.NewHTTPClient(httpAddr)
 
-	if *debugHTTP {
-		testCtx.httpClient.Log = logrus.NewLogger("http-client")
-		testCtx.httpClient.Log.SetLevel(logging.DebugLevel)
-	}
+	// make sure there are no containers left from the previous run
+	resetVppAgents(t, testCtx.dockerClient)
+	resetMicroservices(t, testCtx.dockerClient)
```

Comment:
IMHO the function names are not clear enough what the functions are doing. I would probably use something like removeExistingVPPAgents/removeRunningVPPAgents or something like that. It is longer, but more obvious.

This is on my edge between personal code preferences and constructive code review so feel free to ignore this if you like.